fiddler/tests/test_fiddler.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any, Callable, Dict
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.stubs.aggregator import AggregatorStub
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.fiddler import FiddlerCheck
+
+
+def test_check(dd_run_check, aggregator, instance):
+    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
+    check = FiddlerCheck('fiddler', {}, [instance])
+    dd_run_check(check)
+
+    aggregator.assert_metric('fiddler.accuracy')
+#    aggregator.assert_metric('fiddler.prediction_drift')
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+
+def test_emits_critical_service_check_when_service_is_down(dd_run_check, aggregator, instance):
+    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
+    check = FiddlerCheck('fiddler', {}, [instance])
+    dd_run_check(check)
+    aggregator.assert_service_check('fiddler.can_connect', FiddlerCheck.OK)
```

Comment:
If you want to create unit tests, you could mock your client library to make it return a specific file that you can add to a `fixtures` folder for instance. It might be easier for you to test without relying on the API and make your test deterministic. 

You can find some example in other integrations:
- https://github.com/DataDog/integrations-extras/blob/eefeb80bbe00afd5559c217ab7c45bcc4a53173c/open_policy_agent/tests/test_open_policy_agent.py#L16-L32 here they mocked the `requests.get` function.

Reply:
@FlorentClarret will keep this in the back of my pocket right now. Its much more easier for me to hit the real end point and get the responses than mocking it. Thanks for the pointer anyway. Will sure be handy at some point. :)